phoenix-core/src/main/java/org/apache/phoenix/util/ServerUtil.java

```diff
@@ -66,7 +67,9 @@ public static IOException createIOException(String msg, Throwable t) {
         } else if (t instanceof IOException) {
             // If the IOException does not wrap any exception, then bubble it up.
             Throwable cause = t.getCause();
-            if (cause == null || cause instanceof IOException) {
+            if (cause instanceof RetriesExhaustedWithDetailsException)
```

Comment:
I had to make this change to get TxWriteFailureIT to work correctly. 
If a write fails it ends up eventually throwing a RetriesExhaustedWithDetailsException.